boards/seeed/xiao_rp2040/xiao_rp2040.dts

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2021 Yonatan Schachter
+ * 2023 Seeed Studio inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include <rpi_pico/rp2040.dtsi>
+#include "xiao_rp2040-pinctrl.dtsi"
+#include "seeed_xiao_connector.dtsi"
+#include <freq.h>
+#include <zephyr/dt-bindings/led/led.h>
+#include <zephyr/dt-bindings/pwm/pwm.h>
+
+/ {
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,flash-controller = &ssi;
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,code-partition = &code_partition;
+	};
+
+	aliases {
+		watchdog0 = &wdt0;
+		led-strip = &ws2812;
+		pwm-led0 = &pwm_led0;
+		led0 = &blue_led;
+		led1 = &green_led;
+		led2 = &red_led;
+	};
+
+	pwm_leds {
+	compatible = "pwm-leds";
+	status = "disabled";
+	pwm_led0: pwm_led_0 {
+		pwms = <&pwm 9 PWM_MSEC(20) PWM_POLARITY_INVERTED>;
+		label = "PWM_LED";
+	};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		blue_led: blue_led {
+			gpios = <&gpio0 25 GPIO_ACTIVE_LOW>;
+			label = "BLUE_LED";
+		};
+
+		green_led: green_led {
+			gpios = <&gpio0 16 GPIO_ACTIVE_LOW>;
+			label = "GREEN_LED";
+		};
+
+		red_led: red_led {
+			gpios = <&gpio0 17 GPIO_ACTIVE_LOW>;
+			label = "RED_LED";
+		};
+	};
+};
+
+&pwm {
+pinctrl-0 = <&pwm_ch4b_default>;
+pinctrl-names = "default";
+divider-int-0 = <255>;
+};
+
+&flash0 {
+	/*
+	 * 8MB of flash minus the 0x100 used for
+	 * the second stage bootloader
+	 */
+	reg = <0x10000000 DT_SIZE_M(2)>;
```

Comment:
this is not 8MiB

Reply:
Sorry the comment is wrong, the board has 2MB flash. I'll update that.